week-3/Homework/mandatory/5-journey-planner.js

```diff
@@ -1,25 +1,154 @@
+  - If it does exist, return the index of it, if not return "Not found"
+
+ Hint: search for string methods like Includes and IndexOf.
+*/
+
+function checkCodeIsThere(stringText) {
+  let magicWord = "code";
+  //edit code below
+  if (stringText.includes(magicWord)) {
+    return stringText.indexOf(magicWord);
+  } else {
+    return "Not found";
+  }
+}
+
 /*
   I am new to London and would like to know what transport I can take to different famous locations.
-  An array with London locations have been provided.
+  The input provided contains a list of locations in London. Each of locations is followed by a list
+  of transport modes that can be used to get there.
+
+  Let's see an example:
+  
+  To take to Tower Bridge, you can use tube or river boat. This information will represented as 
+    ["Tower Bridge", "tube", "river boat"]
 
-  Return an array of where I can go if I only want to use a specific mode of transport.
+  Where
+    the 1st element says the name of the location,
+    and rest of them says the transport modes.
 
-  NOTE: only the names should be returned, not the means of transport.
+  You will then get a list of these information, e.g:
+  [
+    ["Tower Bridge", "tube", "river boat"],
+    ["Abbey road", "double decker"],
+    ["London Eye", "tube", "river boat", "bus"]
+  ]
+
+  You have to finish up the body of journeyPlanner function that should tell where I can go if I only
+  want to use a specific mode of transport. But before jumping straight to the main function, we will
+  break down the whole task into smaller steps that make our job easier.
+
+  This technic is also referred as problem decomposition. It helps you to reduce scope of the problem 
+  by only focusing on a small chunk of the whole problem at a time.)
+*/
+
+/*
+  Implement the function getTransportModes that
+   - Accepts an array containing the location and available transport modes
+     e.g: ["Tower Bridge", "tube", "river boat"]
+   - Returns an array including the available transport modes to the given location
+     e.g: ["tube", "river boat"]
+  
+  Hint: Use the corresponding array method to split the array.
 */
+function getTransportModes(transport) {
+  var possibleRoads = transport.slice(1,3)
+  return possibleRoads;
+ }
 
-function journeyPlanner() {
+/*
+  Implement the function isAccessibleByTransportMode that
+   - Accepts two parameters:
+     1) First parameter is an array of transport modes
+        e.g: ["tube", "river boat"]
+     2) Second parameter is a string containing a transport mode
+        e.g: "river boat"
+     
+   - Returns 
+     * True if the location in the first parameter is accessible by the transport mode given in second parameter
+     * Otherwise, returns false
 
+  Hint: Use the corresponding array method to decide if an element is member of an array.
+*/
+function isAccessibleByTransportMode(mode, strMode) { 
+  if (mode.includes(strMode)){
+    return true 
+  } else {
+    return false
+  }
```

Comment:
In this exercise, the inclusdes() true for you no need to write if/else statement. Could you fix that please.

Reply:
done